aws-api/src/main/java/com/amplifyframework/api/aws/GsonUtil.java

```diff
@@ -75,4 +90,80 @@ private static Object toObject(JsonElement element) {
         }
         return null;
     }
+
+    public static Gson getGson() {
+        if (gson == null) {
+            gson = new GsonBuilder()
```

Comment:
Also, if we make the `GsonFactory`, can we extract the logic at L51 of `GsonGraphQLResponseFactory`? Can both places share a single, externally-defined/construted `Gson` instance?

Reply:
I did try this idea at first - I created one `Gson` instance with all the serializers.  I ended up getting a `JsonTreeOverflow` exception because the `GsonListDeserializer` was calling itself recursively if the `Gson` instance inside it called itself.    I've now realized that GsonListDeserializer isn't actually needed, so I deleted it, and now there's only one `Gson` deserializer instance created in `GsonGraphQLResponseFactory`, and one `Gson` serializer instance created in `GsonVariablesSerializer`.